app/models/miq_worker.rb

```diff
@@ -316,19 +316,41 @@ def queue_name
     end
   end
 
-  def self.build_command_line(guid)
-    command_line = "#{Gem.ruby} #{runner_script} --heartbeat --guid=#{guid} #{name}"
-    ENV['APPLIANCE'] ? "nice #{nice_increment} #{command_line}" : command_line
+  def self.build_command_line(options)
+    raise ArgumentError, "expected options hash, received: #{options.class}" unless options.kind_of?(Hash)
+    raise ArgumentError, "missing :guid options key" unless options.key?(:guid)
+
+    prefix = command_line_env_prefix(options)
+    cmd    = "#{prefix}#{Gem.ruby} #{runner_script} --heartbeat #{name}"
+    ENV['APPLIANCE'] ? "nice #{nice_increment} #{cmd}" : cmd
   end
 
+  # Converts:
+  # { :guid   => 8d473240-efb9-11e5-92d7-0050569c24ad,
+  #   :ems_id => 1 }
+  #
+  # To: 'GUID=8d473240-efb9-11e5-92d7-0050569c24ad EMS_ID=1 '
+  def self.command_line_env_prefix(options)
+   return "" unless options.kind_of?(Hash)
+
+    options.each_with_object("") do |kv, prefix|
+      prefix << "#{kv.first.to_s.upcase}=#{kv.last} "
+    end
+  end
+  private_class_method :command_line_env_prefix
```

Comment:
Do you think we need to shellescape anything to be on the safe side??

Reply:
Yeah, I'll try it out.

Reply:
This is now done at the end of `build_command_line`.